### PR TITLE
feat(suite): Update bar traffic lights on mac

### LINKED ISSUE #11399: Mac app top bar improvement
We have a fixed top bar on Mac now, which is sub-optimal.

Please incorporate the top bar into the app for a seamless design experience.
Mind resizing of the menu from #11400 

Screenshots:
Mac:
Screenshot 2024-05-28 at 07 52 49

Windows:
Screenshot 2024-05-28 at 07 53 31

Design: [Figma link](https://www.figma.com/file/KOTm1scRKiHBRTvMgEFhe6/OS-Specifics?type=design&node-id=4002%3A7181&mode=dev&t=NKffSnRun7Y4dhh8-1)

_In the design, there are also the browser and Windows options shown (Linux should keep its current top bad similarly as Windows). Those should basically stay as they are._

### PR DESCRIPTION
**Pull request by jvaclavik**
## Description

updated in
- sidebar
- switch device modal
- onboarding
- logged out screen
- mobile layout

## Related Issue

Resolve #11399

## Screenshots:

Mac:
Screenshot 2024-07-11 at 10 37 11

Screenshot 2024-07-11 at 10 37 26

Linux (no change):
image

### CODE REVIEW

**Comment by jvaclavik** [comment]
@bosomt could you please test it on Windows?

We tested it on Mac and Linux. 

Basically navbar should remain the same on Linux and Windows.

File: packages/suite/src/components/suite/layouts/SuiteLayout/Sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -6,15 +6,23 @@ import { AccountsMenu } from 'src/components/wallet/WalletLayout/AccountsMenu/Ac
...
 
-const Container = styled.nav<{ $elevation: Elevation }>`
+const Container = styled.nav<{ $elevation: Elevation; isMac?: boolean }>`
     display: flex;
     flex-direction: column;
     flex: 0 0 auto;
     height: 100%;
     background: ${mapElevationToBackground};
     border-right: 1px solid ${mapElevationToBorder};
+    ${({ isMac }) => isMac && `padding-top: ${spacingsPx.xl};`}
```
Comment (peter-sanderson) [inline packages/suite/src/components/suite/layouts/SuiteLayout/Sidebar/Sidebar.tsx]: Maybe add a comment that is because o traffic lights

**Review by peter-sanderson — APPROVED** [review]

File: packages/suite/src/components/suite/layouts/SuiteLayout/Sidebar/Sidebar.tsx [hunk 2]
```diff
@@ -6,15 +6,25 @@ import { AccountsMenu } from 'src/components/wallet/WalletLayout/AccountsMenu/Ac
...
 import { SIDEBAR_WIDTH_NUMERIC } from 'src/constants/suite/layout';
-import { Elevation, mapElevationToBackground, mapElevationToBorder, zIndices } from '@trezor/theme';
+import {
+    Elevation,
+    mapElevationToBackground,
+    mapElevationToBorder,
+    spacingsPx,
+    zIndices,
+} from '@trezor/theme';
+import { isMacOs } from '@trezor/env-utils/lib';
```
Comment (tomasklim) [inline packages/suite/src/components/suite/layouts/SuiteLayout/Sidebar/Sidebar.tsx]: Probably wrong import

**Review by tomasklim — CHANGES_REQUESTED** [review]

File: packages/suite-desktop-core/src/app.ts [hunk 3]
```diff
@@ -33,6 +33,8 @@ const createMainWindow = (winBounds: WinBounds) => {
         height: winBounds.height,
         minWidth: MIN_WIDTH,
         minHeight: MIN_HEIGHT,
+        titleBarStyle: 'hidden',
```
Comment (tomasklim) [inline packages/suite-desktop-core/src/app.ts]: according to docs, it is also available on windows

Comment (STew790) [reply packages/suite-desktop-core/src/app.ts]: there's no titleBar on Windows, unfortunately
image

Comment (jvaclavik) [reply packages/suite-desktop-core/src/app.ts]: It should be fixed now

**Review by tomasklim — APPROVED** [review]

File: packages/suite/src/views/suite/SwitchDevice/SwitchDeviceModal.tsx [hunk 4]
```diff
@@ -11,7 +13,7 @@ type SwitchDeviceModalProps = {
     isAnimationEnabled?: boolean;
 };
 
-const Container = styled.div`
+const Container = styled.div<{ $hasTopPadding?: boolean }>`
```
Comment (peter-sanderson) [inline packages/suite/src/views/suite/SwitchDevice/SwitchDeviceModal.tsx]: nit: extract type

File: packages/suite/src/components/suite/TrafficLightOffset.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import React from 'react';
+import styled from 'styled-components';
+
+type Props = {
+    children?: React.ReactNode;
+    offset?: number;
+};
+
+// this element fixes not being able to drag window: https://github.com/electron/electron/issues/5678
+const WindowAppHandler = styled.div`
```
Comment (peter-sanderson) [inline packages/suite/src/components/suite/TrafficLightOffset.tsx]:
```suggestion
// See: https://github.com/electron/electron/issues/5678
const FixForNotBeingAbleToDragWindow = styled.div`
```

Better to incorporate the information into naming rather then comments as they tend to decay/being not-updated.

Also this is then clear at place where this is used.

**Review by peter-sanderson — APPROVED** [review]
Just small stuff. :ship:

**Pull request merged**